wgengine/magicsock/magicsock.go

```diff
@@ -3630,6 +3633,20 @@ func (de *endpoint) cliPing(res *ipnstate.PingResult, cb func(*ipnstate.PingResu
 }
 
 func (de *endpoint) send(b []byte) error {
+	// if heartbeat disabled, try using alternate path
+	if de.heartbeatDisabled {
+		if !de.pathFinderRunning {
+			de.startPathFinder()
+		}
+		if sendFunc, ok := de.atomicSenderFunc.LoadOk(); ok {
```

Comment:
I forgot that our syncs.AtomicValue does actually transition from nil to non-nil to nil again because it always stores typed nils in the interface. So this LoadOk isn't needed. We can use a nil func pointer as the sentinel value for "need to take the slow path"